toolchain/parse/handle_var.cpp

```diff
@@ -23,7 +23,11 @@ static auto HandleVar(Context& context, State finish_state,
     context.AddLeafNode(NodeKind::ReturnedModifier, returned_token);
   }
 
-  context.PushState(State::BindingPatternAsVariable);
+  if (context.PositionKind() == Lex::TokenKind::OpenParen) {
+    context.PushState(State::ParamListAsRegular);
+  } else {
+    context.PushState(State::BindingPattern);
+  }
```

Comment:
I wonder if it's worth factoring out either a function or a state to do this. `State::Pattern`?

Reply:
Good idea; done.